cluster/network.go

```diff
@@ -78,6 +78,34 @@ func (networks Networks) Filter(names []string, ids []string, types []string) Ne
 	return out
 }
 
+// RemoveDuplicateEndpoints returns a copy of input network
+// where duplicate container endpoints in the network are removed.
+// See https://github.com/docker/swarm/issues/1969
+// This function should be disabled when the bug is fixed in Docker network
+func (network *Network) RemoveDuplicateEndpoints() *Network {
+	// build a map from endpointID -> endpointIndex
+	endpointMap := make(map[string]string)
+	// traverse the endpoints to find the correct endpointIndex for each endpointID
+	for endpointIndex, endpointResource := range network.NetworkResource.Containers {
+		endpointID := endpointResource.EndpointID
+		// if this endpointID doesn't exist yet, add it
+		// if this endpointID exists, but endpointIndex is not a duplicate, use
+		// this endpointIndex
+		if _, ok := endpointMap[endpointID]; !ok || !strings.HasPrefix(endpointIndex, "ep-") {
```

Comment:
@dongluochen it would better to check if the `endpointIndex` contains `endpointID` instead of checking for `ep-`. We added the `ep-` prefix mostly for better UX. but for programmatic checks, it is better to check for `endpointID` part of the index.